code/game/machinery/portable_turret.dm

```diff
@@ -701,8 +701,21 @@
 	var/list/targets = list()			//list of primary targets
 	var/list/secondarytargets = list()	//targets that are least important
 
+	var/list/seenturfs = list()
+	for(var/turf/T in oview(world.view, src))
+		seenturfs += T
+
+	for(var/mob in player_list)

```

Comment:
player_list precludes anti-carp turrets (or any other npc, as the case may be) unless the carp is being actively controlled by a client

Reply:
changed to living_mob_list